### PR TITLE
sql: adding new tables at pg_catalog

### LINKED ISSUE #63862: sql: add compatible types to pg_catalog
When we added the new tables/columns to `pg_catalog`, we ignored any column where the type OID from Postgres did not exist in CockroachDB. One example is OID 194, which in postgres is `pg_node_tree`. However, under the hood, a `pg_node_tree` is the same as a normal string.

So we actually can add these columns in CockroachDB and just use a string type. (Actually, for some of the existing pg_catalog tables, we already did this.)

Tasks:
- [ ] Check all the pg_catalog columns that have an OID of 194, and add those columns to CockroachDB if they are missing.
- [ ] Identify other OIDs in the Postgres pg_catalog that are identical to strings.
- [ ] Add these other columns to CockroachDB.
- [ ] Also, I think the Postgres `yes_or_no` type can be treated as a boolean in CockroachDB.

### PR DESCRIPTION
**Pull request by mnovelodou**
Previously, we were unable to add this tables because its
types were unimplemented
This was inadequate because some these unimplemented types
could be mapped to other implemented types
To address this, this patch adds these tables by mapping
unimplemented types.

Release note (sql): Added new tables at pg_catalog:
- pg_partitioned_table
- pg_replication_origin_status
- pg_init_privs
- pg_replication_slots
- pg_policy
- pg_sequences
- pg_subscription_rel
- pg_largeobject_metadata

### CODE REVIEW

**Comment by mnovelodou** [comment]
Will merge after mappings on PR #63963 is approved.

**Review by rafiss — APPROVED** [review]
this looks good! thank you

one question though; aren't there some existing tables that currently have columns missing that we can add now?

**Comment by rafiss** [comment]
this looks good!

bors r+

**Pull request merged**
